libvcx/src/aries/handlers/connection/connection.rs

```diff
@@ -262,7 +269,15 @@ impl Connection {
      */
     pub fn connect(&mut self) -> VcxResult<()> {
         trace!("Connection::connect >>> source_id: {}", self.source_id());
-        self.step(DidExchangeMessages::Connect())
+        self.connection_sm = match &self.connection_sm {
+            SmConnection::Inviter(sm_inviter) => {
+                SmConnection::Inviter(sm_inviter.clone().transition_connect()?)
+            }
+            SmConnection::Invitee(sm_invitee) => {
+                SmConnection::Invitee(sm_invitee.clone().transition_connect()?)
+            }
+        };
```

Comment:
Maybe it would be nice if inviter / invitee SMs implemented a trait and `Connection` would reference an implementor of the trait, so that we may avoid these branches